### PR TITLE
Cleaning up Ruby warnings and deprecations

### PR DESCRIPTION
**Pull request by acant**
So I started out cleaning up the deprecated Fixnum class...
and then I kept going and clean up some other Ruby warnings...
and now those are cleaned up I added the warnings flag into the Minitest and RSpec tasks.

None of changes were very significant, except for updating the names of the Minitest classes to make them consistent. The scheme I choose was to make the class names match the file names.
(e.g., FakeJoin to FakeFileJoinTest), but if that is not the best choice I am happy to change to another scheme.

### CODE REVIEW

File: lib/fakefs/file.rb [hunk 1]
```diff
@@ -280,14 +280,14 @@ def self.executable?(filename)
     def self.chown(owner_int, group_int, filename)
       file = FileSystem.find(filename)
       if owner_int && owner_int != -1
-        owner_int.is_a?(Fixnum) || fail(TypeError,
-                                        "can't convert String into Integer")
-        file.uid = owner_int
+        owner_int.respond_to?(:to_i) || fail(TypeError,
```
Comment (grosser) [inline lib/fakefs/file.rb]:
```
''.respond_to?(:to_i)
=> true
```

... if this did not break anything maybe add the missing test ... can check for `is_a?(Integer)` though ...

Comment (grosser) [inline lib/fakefs/file.rb]: ... just use `Integer(owner_int)` ... that should do it ... and raises a nice error

File: test/fake/file/sysseek_test.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 require_relative '../../test_helper'
 
 # File SysSeek test class
-class FileSysSeek < Minitest::Test
+class FakeFileSysSeekTest < Minitest::Test
```
Comment (grosser) [inline test/fake/file/sysseek_test.rb]: ❤️

**Comment by grosser** [comment]
that $PID fix is a real bug I think ...

File: lib/fakefs/file.rb (hunk 1 above)
Comment (acant) [inline lib/fakefs/file.rb]: I like that option! Particularly that it also eliminates the fail calls.
Give a moment and I'll update this commit.

Comment (acant) [inline lib/fakefs/file.rb]: Updated. I also specified the base in Integer(), which is probably not important for any of the numbers that will realistically pass to this method but feel like it is probably the correct usage.
It was mentioned here, when I will looking up the usage of Integer:
https://moeffju.net/blog/integer-vs-to_i-in-ruby

File: lib/fakefs/file.rb [hunk 3]
```diff
@@ -279,16 +279,8 @@ def self.executable?(filename)
...
-        owner_int.is_a?(Fixnum) || fail(TypeError,
-                                        "can't convert String into Integer")
-        file.uid = owner_int
-      end
-      if group_int && group_int != -1
-        group_int.is_a?(Fixnum) || fail(TypeError,
-                                        "can't convert String into Integer")
-        file.gid = group_int
-      end
+      file.uid = Integer(owner_int, 10) if owner_int && owner_int != -1
```
Comment (grosser) [inline lib/fakefs/file.rb]: feels weird that we support 2 types of disabling ...

Comment (grosser) [inline lib/fakefs/file.rb]: (old behavior so whatever ... )

**Comment by acant** [comment]
ooooo so my conversion of the #to_i into Integer() fails sort of spectacularly on various versions. I think I re-based in haste. :) It is the end of my day right now, but I'll investigate and fix this up tomorrow.
@grosser thanks for you feedback! It has been really helpful.

File: lib/fakefs/file.rb (hunk 3 above)
Comment (grosser) [inline lib/fakefs/file.rb]: actually this is wrong ... we cannot convert string to integer without changing behavior ... and behavior should be based on original chown ... so either make sure it works the same in the original or use `.is_a?(Integer)`

Comment (acant) [inline lib/fakefs/file.rb]: I think your right. I will push a commit tomorrow to use .is_a?(Integer).

**Comment by grosser** [comment]
thx! 0.11.1 is out!

**Comment by acant** [comment]
Aweome! Thanks again for your feedback.

**Pull request merged**